pkg/apis/app/v1/kieapp_types.go

```diff
@@ -23,14 +23,49 @@ type KieAppSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// KIE environment type to deploy (prod, authoring, trial, etc)
-	Environment    string           `json:"environment,omitempty"`
+	Environment    EnvironmentType  `json:"environment,omitempty"`
```

Comment:
Mostly a question about the pros/cons of this. I think the advantage, type safety, is obvious and I like it. However, did we previously have an option for users to add their own ad-hoc named environments, and do we lose that?

I can see in the rest of the PR that naming environments brings a lot of benefits, and think it's absolutely acceptable to make this trade-off, but would just want us to make the decision consciously.

Reply:
As it is still a string, it is possible to add more, see the unit tests